File: DBGenerator/DbContainer/DbRow.cs
using System;
using System.Collections.Generic;

namespace DBContainer
{

    public partial class DbRow
    {
        public DbTable          parent          = null;
        public int              rowIndex        = 0;

        public DbValue this[ int colIdx ]
        {
            get
            {
                return new DbValue { parent = this, columnIndex = colIdx };
            }
        }
        public DbValue this[ string colName ]
        {
            get
            {
                return new DbValue { parent = this, columnIndex = parent.getIndexByColumnName( colName ) };
            }
        }

    }

}

File: xcode_cmd_test1/Lib/DbRow.cpp
#include "All.h"

namespace xxx
{
    DbValue DbRow::operator[]( int colIdx )
    {
        return DbValue{ this, colIdx };
    }
    DbValue DbRow::operator[]( String const& colName )
    {
        return DbValue{ this, parent->GetIndexByColumnName( colName ) };
    }
    DbValue DbRow::operator[]( DbColumn const& col )
    {
        return DbValue{ this, col.columnIndex };
    }
}

